src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -218,6 +220,13 @@ abstract class Model implements Arrayable, ArrayAccess, CanBeEscapedWhenCastToSt
      */
     protected static string $builder = Builder::class;
 
+    /**
+     * The Eloquent collection class to use for the model.
+     *
+     * @var class-string<\Illuminate\Database\Eloquent\Collection<*, *>>
+     */
+    protected static string $collection = Collection::class;
```

Comment:
Unfortunately this property name is already used by a popular third party package. I opened a PR to propose a new name: https://github.com/laravel/framework/pull/52186

Reply:
I wasn't aware, thanks!

Reply:
No problem, I hope the patch will be accepted before this feature is released.

Reply:
Just out of curiosity, why don't they just use the existing `$table` property for that purpose?

Reply:
In the MongoDB terminology, a collection is the equivalent of an SQL table (see [this doc](https://www.mongodb.com/docs/manual/reference/sql-comparison/).

This property was introduced [a long time](https://github.com/mongodb/laravel-mongodb/commit/7effee857b64b2fb1d499e2452446661d568a7bf) before I start working on the project. I think they wanted to use the terms in the code.